### PR TITLE
Add nyaml2nxdl updates from NIAC PR1303

### LINKED ISSUE #96: Fetching changes from NAIC PR1303 to FAIRmat
Fetch code from NIAC PR [1303](https://github.com/nexusformat/definitions/pull/1303) into fairmat branch.
- Rebase on `fairmat` branch
- Modify small keywords that are related with branches so that the ci/cd pass properly.

### PR DESCRIPTION
**Pull request by RubelMozumder**
- [x] Bring code modification from the `nyam2nxd-migration` branch (in PR for NIAC).

Fixes #96

### CODE REVIEW

**Comment by RubelMozumder** [comment]
@domna, this is the separate PR for the code modification nyam2nxdl-migration.

**Comment by domna** [comment]
This should be merged as soon as https://github.com/nexusformat/definitions/pull/1303 is accepted

**Review by domna — CHANGES_REQUESTED** [review]
Something is wrong here. The Makefile has duplicate build instructions which are not in the NIAC PR.

**Comment by RubelMozumder** [comment]
> Something is wrong here. The Makefile has duplicate build instructions which are not in the NIAC PR.

Can you refer the lines: 
I checked the highlighted code (green color) theses are coming from either NIAC or Fairmat. I do  not GH is behaving weirdly. Because the part that exists in the Fairmat branch is also showing in green highlighted.

File: Makefile [hunk 1]
```diff
@@ -110,11 +107,23 @@ $(APPDEF_DIR)/%.nxdl.xml : $(APPDEF_DIR)/$(NYAML_SUBDIR)/%.yaml
...
+$(CONTRIB_DIR)/%.nxdl.xml : $(CONTRIB_DIR)/$(NYAML_SUBDIR)/%.yaml
+	$(PYTHON) -m dev_tools.nyaml2nxdl.nyaml2nxdl --input-file $<
+	mv $(CONTRIB_DIR)/$(NYAML_SUBDIR)/$*.nxdl.xml $@
+
+$(APPDEF_DIR)/%.nxdl.xml : $(APPDEF_DIR)/$(NYAML_SUBDIR)/%.yaml
+	$(PYTHON) -m dev_tools.nyaml2nxdl.nyaml2nxdl --input-file $<
+	mv $(APPDEF_DIR)/$(NYAML_SUBDIR)/$*.nxdl.xml $@
+
+NXDLS := $(NXDL_APPDEF) + $(NXDL_CONTRIB) + $(NXDL_BC)
+nyaml :
```
Comment (domna) [inline Makefile:123]: These are fine, but here are also lines 98-108 which use the `nyaml2nxdl` command for the same build command.

Comment (RubelMozumder) [reply Makefile:123]: I see. That part was included in rebase. That should be removed.

File: pyproject.toml
Comment (domna) [inline pyproject.toml:1]: This was removed in the NIAC PR. Why is it still here?

**Comment by domna** [comment]
Closing this in favor of #102

**Pull request closed without merging**